### PR TITLE
Support namespace in path for requesting k8s api from web console

### PR DESCRIPTION
**Pull request by jwforres**

### CODE REVIEW

**Comment by jwforres** [comment]
@liggitt review

v1beta3 k8s api was not supposed to be allowing namespace as a query param anymore.  After the next rebase it may no longer work.  osapi does not  yet support namespace in path.  So for now we have to handle namespace in two different ways depending on which api we are pointing to and whether we are building a URL that includes a namespace or not.

@deads2k @ncdc when you guys do the next rebase, the only change you will need to make in the console after this is to switch the URL templates at the bottom of service/data.js from /ns/{namespace}/ to /namespaces/{namespace}/  (and rebuild bindata.go)

File: assets/app/scripts/services/data.js [hunk 1]
```diff
@@ -605,15 +613,21 @@ angular.module('openshiftConsole')
       protocol = window.location.protocol === "http:" ? "http" : "https"; 
     }
 
+    var namespaceInPath = params.namespace && SERVER_TYPE_MAP[type].namespacePath;
+    var namespace = null;
+    if (namespaceInPath) {
+      namespace = params.namespace;
+      params.namespace = null;
```
Comment (liggitt) [inline assets/app/scripts/services/data.js]: `delete`?

File: assets/app/scripts/services/data.js [hunk 2]
```diff
@@ -605,15 +613,21 @@ angular.module('openshiftConsole')
       protocol = window.location.protocol === "http:" ? "http" : "https"; 
     }
 
+    var namespaceInPath = params.namespace && SERVER_TYPE_MAP[type].namespacePath;
+    var namespace = null;
+    if (namespaceInPath) {
+      namespace = params.namespace;
```
Comment (liggitt) [inline assets/app/scripts/services/data.js]: How does the URL template behave if we get in here and namespace ends up null or undefined? Should we default to ""?

File: assets/app/scripts/services/data.js (hunk 1 above)
Comment (liggitt) [inline assets/app/scripts/services/data.js]: Do we need to worry about mutating params? Not sure what calls this and if it expects namespace to be unset on the params object it passes in

Comment (liggitt) [inline assets/app/scripts/services/data.js]: Looks like it's safe to mutate, but I'd prefer we do
```
params = angular.copy(params);
delete params.namespace;
```

**Comment by jwforres** [comment]
@liggitt updated with review feedback and bindata generated

**Comment by liggitt** [comment]
LGTM

**Comment by jwforres** [comment]
[merge]

**Pull request merged**